app/Models/DocuSignEnvelope.php

```diff
@@ -244,4 +248,25 @@ public function getSensibleExtractionUrlAttribute(): ?string
             null :
             'https://app.sensible.so/extraction/?e='.$this->sensible_extraction_uuid;
     }
+
+    public static function getEnvelopeUuidFromSummaryPdf(string $summary_pdf_text): string
+    {
+        $summary_text = (new Parser())
+            ->parseContent($summary_pdf_text)
+            ->getText();
+
+        $matches = [];
+
+        if (preg_match(self::ENVELOPE_ID_REGEX, $summary_text, $matches) !== 1) {
+            throw new \Exception('Could not extract envelope ID');
+        }
+
+        return Str::lower(
+            Str::substr($matches['envelopeId'], 0, 8).'-'.
+            Str::substr($matches['envelopeId'], 8, 4).'-'.
+            Str::substr($matches['envelopeId'], 12, 4).'-'.
+            Str::substr($matches['envelopeId'], 16, 4).'-'.
+            Str::substr($matches['envelopeId'], 20, 12)
```

Comment:
Why not use the regex to extract the fields individually instead of using substr calls? I'm not going to reject the PR over this. It just seems silly.

Reply:
I don't think that's substantially better.

Reply:
Fair enough.